### DISCUSSION TITLE
UnboundLocalError when calling beeswarm

### DISCUSSION THREAD

**Discussion by manavka** [shap/shap#3166] [Q&A]
Hello everyone,

I am currently experimenting with beeswarms vs summary plots. I am able to call shap.summary_plot(shap_values,X_train.columns) with no issues. The figure is correctly outputted and the features are all labeled. However, when I try to call shap.plots.beeswarm(shap_values,X_train.columns) or shap.plots.beeswarm(shap_values), I get the attached UnboundLocalError. Does anyone know why this is occurring? The examples in the docs show them calling the beeswarm with just the shap values so I don't think the issue is a lack of arguments being passed to the function.
![Error Message](https://github.com/shap/shap/assets/71353846/2d1c534a-4da8-4262-807d-45dd3c44d5e2)

**Comment by thatlittleboy** [comment]
@manavka What version of `shap` are you using? Try updating to the latest version first.
